### PR TITLE
Make Babybear field FFT friendly

### PR DESCRIPTION
**Pull request by mdvillagra**
# Make Babybear field FFT friendly

## Description

Implemented `isFFT`trait to babybear field as requested in issue #539. Here the constants`TWO_ADIC_PRIMITVE_ROOT_OF_UNITY` and `TWO_ADICITY` were updated accordingly.
## Type of change

- [X] New feature

## Checklist
- [X] Linked to Github Issue
- [X] Unit tests added

### CODE REVIEW

File: math/src/field/fields/fft_friendly/babybear.rs [hunk 1]
```diff
@@ -98,4 +116,20 @@ mod test_babybear_31_bytes_ops {
         let from_bytes = FieldElement::<Babybear31PrimeField>::from_bytes_be(&bytes).unwrap();
         assert_eq!(element, from_bytes);
     }
+
```
Comment (diegokingston) [inline math/src/field/fields/fft_friendly/babybear.rs:119]: There should be a test using FFT to show that everything's working properly. Also, use cargo fmt to help with the code formatting

**Review by diegokingston — CHANGES_REQUESTED** [review]

File: math/src/field/fields/fft_friendly/babybear.rs (hunk 1 above)
Comment (mdvillagra) [reply math/src/field/fields/fft_friendly/babybear.rs:119]: should the FFT test be added inside babybear.rs? or inside polynomial.rs like was done for `Stark252PrimeField`?

Comment (MauroToscano) [reply math/src/field/fields/fft_friendly/babybear.rs:119]: You can add it inside babybear, but under the cfg[test], so it's only part of that module for testing purposes.

**Review by diegokingston — APPROVED** [review]

**Comment by MauroToscano** [comment]
There's a bug with how the CUDA and Metal is used. It's assuming that if the feature is enabled, it can run with CUDA or Metal. As a quickfix, you can add this in the relevant tests`#[cfg(not(feature = "instruments"))]`. We should fix this in more general way, even if the feature is not enabled it could backoff to the CPU version, but it's not critical.

**Review by MauroToscano — APPROVED** [review]

**Review by Oppen — APPROVED** [review]

**Pull request merged**